apps/prairielearn/src/ee/lib/ai-grading.sql

```diff
@@ -97,13 +99,14 @@ WITH
   )
 SELECT
   emb.submission_text,
-  ls.iq_score_perc AS score_perc,
-  ls.iq_id AS instance_question_id
+  lgj.iq_score_perc AS score_perc,
+  lgj.iq_id AS instance_question_id
 FROM
-  latest_submissions ls
-  JOIN submission_grading_context_embeddings AS emb ON (emb.submission_id = ls.s_id)
+  latest_grading_jobs lgj
+  JOIN submission_grading_context_embeddings AS emb ON (emb.submission_id = lgj.s_id)
 WHERE
-  ls.rn = 1
+  lgj.rn = 1
+  AND NOT lgj.is_ai_graded
```

Comment:
There are two ways we could implement filtering here:

- For each instance question, use the latest non-AI-graded grading job
- For each instance question, use the latest grading job only if it is non-AI-graded

These are subtly different. You implemented the second option, is that the one we want here? Not asking for a change, only a confirmation that you've thought this through.

Reply:
Yes. The second is what I've intended to build here.

Reply:
My rationale, just in case people are wondering: 
For manual grades, it is hard for instructors to check the older manual grading results for the same submission once it is regraded (from what I've tried with manual grading), so the quality of non-latest grading jobs are not guaranteed. If I give a random and bad grade to a submission first, and then overwrite it with an AI grade, the older bad grade would be used if I'm using the latest non-AI-graded job.